src/core/Component.js

```diff
@@ -0,0 +1,49 @@
+export default class Component {
+  $target;
+  state;
+  router;
+
+  constructor($target, router) {
+    //인스턴스 생성시 실행.
+    this.$target = $target;
+    this.setup();
+    this.render();
+    this.setEvent();
+    this.router = router;
```

Comment:
P555;

정말 소소하긴 하지만 혹시모를 상황을 방지하기 위해 순서를 저는 아래와 같이 할 것 같아요.

```suggestion
    this.$target = $target;
    this.router = router;
    
    this.setup();
    this.render();
    this.setEvent();
```